test/testExtractScripts.js

```diff
@@ -0,0 +1,51 @@
+const assert = require('assert');
+const fs = require('fs');
+const tmp = require('tmp');
+
+const extractScriptsFromHTML = require('../src/extractScripts.js');
+
+// IMPORTANT: if file naming convention changes in file, change this function.
+function expectedResults(fileName, dirName, numberOfScriptTags) {
+  const referenceBase = './test/referenceFiles/ref_';
+  const files = [];
+  const referenceFiles = [];
+  for (let i = 0; i < numberOfScriptTags; i++) {
+    const newFileName = 'script' + i + '_' + fileName + '.js';
+    files.push(dirName + '/' + newFileName);
+    referenceFiles.push(referenceBase + newFileName);
+  }
+  return {fileList: files, references: referenceFiles};
+}
+
+describe('#extractScriptsFromHTML()', function() {
+  let tempDir = null;
+  before(function() {
+    tempDir = tmp.dirSync({prefix: 'testTmp', unsafeCleanup: true, tmpdir: '.'});
+  });
+
+  after(function() {
+    tempDir.removeCallback();
+  });
+
+  context('2 sets of <script> tags- file-list-test.html', function() {
```

Comment:
For my own info since I'm new to mocha -- why `context` here instead of `describe`? It sounds like they're equivalent, but is there a preference for using them in a certain way?

Reply:
My understanding is `context` is an alias of `describe`, [here](https://mochajs.org/#bdd) is the documentation mentioning that. Says that `context` "behaves the same way; it provides a way to keep tests easier to read and organized. Similarly, specify() is an alias for it()"